### ISSUE TITLE
JSON validation during serialization not working

### ISSUE THREAD

**Issue by matanmarciano** [confluentinc/confluent-kafka-go#1345]
Description
===========
I am implementing the Kafka producer side with schema registry (JSON)

I have already a schema in the client registry and when I producing a wrong message (completely different), I am not getting any errors during serialization. 

Attached are screenshots of:
- Schema registry with id from my kafka ui
- Debugging break point inside your package shows that the latest schema id is match

Here is my code:
```
package main

import (
	"fmt"
	"os"

	"github.com/confluentinc/confluent-kafka-go/kafka"
	"github.com/confluentinc/confluent-kafka-go/schemaregistry"
	"github.com/confluentinc/confluent-kafka-go/schemaregistry/serde"
	"github.com/confluentinc/confluent-kafka-go/schemaregistry/serde/jsonschema"
)

func main() {

	if len(os.Args) != 4 {
		fmt.Fprintf(os.Stderr, "Usage: %s <bootstrap-servers> <schema-registry> <topic>\n",
			os.Args[0])
		os.Exit(1)
	}

	bootstrapServers := os.Args[1]
	url := os.Args[2]
	topic := os.Args[3]

	p, err := kafka.NewProducer(&kafka.ConfigMap{"bootstrap.servers": bootstrapServers})

	if err != nil {
		fmt.Printf("Failed to create producer: %s\n", err)
		os.Exit(1)
	}

	fmt.Printf("Created Producer %v\n", p)

	client, err := schemaregistry.NewClient(schemaregistry.NewConfig(url))

	if err != nil {
		fmt.Printf("Failed to create schema registry client: %s\n", err)
		os.Exit(1)
	}

	// Set up the JSON schema serializer configuration
	sc := jsonschema.NewSerializerConfig()
	sc.AutoRegisterSchemas = false
	sc.UseLatestVersion = true
	sc.EnableValidation = true

	// Create the JSON schema serializer
	ser, err := jsonschema.NewSerializer(client, serde.ValueSerde, sc)

	if err != nil {
		fmt.Printf("Failed to create serializer: %s\n", err)
		os.Exit(1)
	}

	// Create a delivery channel for producer events
	deliveryChan := make(chan kafka.Event)

	value := User{
		Foo: 42,
		Bar: "blue",
	}

	// Serialize the payload
	payload, err := ser.Serialize(topic, &value)
	if err != nil {
		fmt.Printf("Failed to serialize payload: %s\n", err)
		os.Exit(1)
	}

	// Produce the message
	err = p.Produce(&kafka.Message{
		TopicPartition: kafka.TopicPartition{Topic: &topic, Partition: kafka.PartitionAny},
		Value:          payload,
	}, deliveryChan)
	if err != nil {
		fmt.Printf("Produce failed: %v\n", err)
		os.Exit(1)
	}

	// Handle the delivery confirmation
	e := <-deliveryChan
	m := e.(*kafka.Message)

	if m.TopicPartition.Error != nil {
		fmt.Printf("Delivery failed: %v\n", m.TopicPartition.Error)
	} else {
		fmt.Printf("Delivered message to topic %s [%d] at offset %v\n",
			*m.TopicPartition.Topic, m.TopicPartition.Partition, m.TopicPartition.Offset)
	}

	close(deliveryChan)
}

// User struct intentionally missing the 'name' field
type User struct {
	Foo int64  `json:"bar"`
	Bar string `json:"foo"`
}

```

<img width="857" alt="image" src="https://github.com/user-attachments/assets/fa7ecbda-30e5-428e-9b5b-bf7cb7b0d4c8">

<img width="1226" alt="image" src="https://github.com/user-attachments/assets/7e06fe23-e446-48f6-b021-1cee646225a2">

confluent-kafka-g version: 1.9.2

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-go and librdkafka version (`LibraryVersion()`):
 - [ ] Apache Kafka broker version:
 - [x] Client configuration: `ConfigMap{...}`
 - [ ] Operating system:
 - [ ] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by rayokota** [comment]
@matanmarciano , please try the latest version 2.6.1

**Comment by matanmarciano** [comment]
@rayokota thanks, solved by upgrading to the latest version.

Actually, it was fixed since [v2.4.0](https://github.com/confluentinc/confluent-kafka-go/releases/tag/v2.4.0)
